demes/demes.py

```diff
@@ -651,26 +1109,81 @@ def isclose(
         :rtype: bool
         """
 
-        def sorted_eq(aa, bb, *, rel_tol, abs_tol) -> bool:
+        def sorted_eq(aa, bb, *, rel_tol, abs_tol, name) -> bool:
             # Order-agnostic equality check.
-            if len(aa) != len(bb):
-                return False
+            assert len(aa) == len(bb)
             for (a, b) in zip(sorted(aa), sorted(bb)):
-                if not a.isclose(b, rel_tol=rel_tol, abs_tol=abs_tol):
-                    return False
+                try:
+                    a.assert_close(b, rel_tol=rel_tol, abs_tol=abs_tol)
+                except AssertionError as e:
+                    raise AssertionError(f"Failed for {name} {aa} and {bb}") from e
```

Comment:
I guess this should instead be `Failed for {name} {a} and {b}`?

Reply:
`a` and `b` is elements of lists `aa` and `bb`, are you sure that you want to see element not the whole lists?

Reply:
Maybe something like `Failed for {name} {aa} and {bb} (elements {a} and {b})`?

Reply:
At the moment, this prints out the lists of demes, so its not clear which deme is causing the difference. E.g. if I run the script I posted earlier in https://github.com/grahamgower/demes/pull/139#issuecomment-734159277 (with the obvious changes), this is the output:
```
Traceback (most recent call last):
  File "/home/grg/src/demes/demes/demes.py", line 956, in assert_close
    e1.assert_close(e2, rel_tol=rel_tol, abs_tol=abs_tol)
  File "/home/grg/src/demes/demes/demes.py", line 198, in assert_close
    assert isclose(
AssertionError: Failed for initial_size 500 != 600 (other).

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/grg/src/demes/demes/demes.py", line 1117, in sorted_eq
    a.assert_close(b, rel_tol=rel_tol, abs_tol=abs_tol)
  File "/home/grg/src/demes/demes/demes.py", line 958, in assert_close
    raise AssertionError(
AssertionError: Failed for epochs (number 1) Epoch(start_time=100, end_time=0, initial_size=500, final_size=500, size_function='constant', selfing_rate=None, cloning_rate=None) and Epoch(start_time=100, end_time=0, initial_size=600, final_size=600, size_function='constant', selfing_rate=None, cloning_rate=None)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/a.py", line 21, in <module>
    g1.assert_close(g2)
  File "/home/grg/src/demes/demes/demes.py", line 1128, in assert_close
    sorted_eq(
  File "/home/grg/src/demes/demes/demes.py", line 1119, in sorted_eq
    raise AssertionError(f"Failed for {name} {aa} and {bb}") from e
AssertionError: Failed for demes [Deme(id='A', description=None, ancestors=None, proportions=None, epochs=[Epoch(start_time=inf, end_time=0, initial_size=100, final_size=100, size_function='constant', selfing_rate=None, cloning_rate=None)]), Deme(id='B', description=None, ancestors=None, proportions=None, epochs=[Epoch(start_time=inf, end_time=100, initial_size=100, final_size=100, size_function='constant', selfing_rate=None, cloning_rate=None), Epoch(start_time=100, end_time=0, initial_size=500, final_size=500, size_function='constant', selfing_rate=None, cloning_rate=None)])] and [Deme(id='A', description=None, ancestors=None, proportions=None, epochs=[Epoch(start_time=inf, end_time=0, initial_size=100, final_size=100, size_function='constant', selfing_rate=None, cloning_rate=None)]), Deme(id='B', description=None, ancestors=None, proportions=None, epochs=[Epoch(start_time=inf, end_time=100, initial_size=100, final_size=100, size_function='constant', selfing_rate=None, cloning_rate=None), Epoch(start_time=100, end_time=0, initial_size=600, final_size=600, size_function='constant', selfing_rate=None, cloning_rate=None)])]
```

Reply:
It could even just print the deme id for the demes being compared. I'll leave it up to you.

Reply:
It looks like it would be better to print it with line break like:
```
Failed for epochs (number 1)
Epoch(start_time=100, end_time=0, initial_size=500, final_size=500, size_function='constant', selfing_rate=None, cloning_rate=None)
Epoch(start_time=100, end_time=0, initial_size=600, final_size=600, size_function='constant', selfing_rate=None, cloning_rate=None)
```
For better quick look

I definitely like `deme.id`  instead the whole deme.